test/common_nn.py

```diff
@@ -322,7 +325,7 @@ def smoothl1loss_reference(input, target, size_average=True, reduce=True):
 loss_reference_fns = {
     'KLDivLoss': kldivloss_reference,
     'NLLLoss': nllloss_reference,
-    'NLLLoss2d': nllloss2d_reference,
+    'NLLLossNd': nlllossNd_reference,
```

Comment:
Did you add `NLLLossNd` anywhere? I can't find the change in `nn`. We might want to deprecate `NLLLoss2d`.

Reply:
NLLLossNd isn't actually a module, I added it to NLLLoss. This line here is a dictionary that maps strings to reference functions for testing.

NLLLoss dispatches to the normal case or the backend's NLLLoss2D (for dimensions >= 2)

Reply:
I see, thanks for the explanation. I still think that if `NLLLoss` now generalizes `NLLLoss2d`, then we should deprecate the latter, because there's no reason to use it anymore.